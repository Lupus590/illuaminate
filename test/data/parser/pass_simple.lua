test["return"] = test.enter --- @local
