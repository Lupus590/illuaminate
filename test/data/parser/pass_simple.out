{ Syntax.program =
  [(Syntax.Assign
      { Syntax.assign_vars =
        (Syntax.SepList1.Mono
           Syntax.NLookup {
             tbl =
             (Syntax.Ref
                (Syntax.NVar
                   (Syntax.Var
                      Node.Node {leading_trivia = []; trailing_trivia = [];
                        contents = "test"; span = pass_simple.lua[1:1-1:5]})));
             open_k =
             Node.Node {leading_trivia = []; trailing_trivia = [];
               contents = [; span = pass_simple.lua[1:5-1:6]};
             key =
             (Syntax.String
                { Syntax.lit_value = "return";
                  lit_node =
                  Node.Node {leading_trivia = []; trailing_trivia = [];
                    contents = "\"return\""; span = pass_simple.lua[1:6-1:14]}
                  });
             close_k =
             Node.Node {leading_trivia = [];
               trailing_trivia =
               [{ Span.value = (Node.Whitespace " ");
                  span = pass_simple.lua[1:15-1:16] }
                 ];
               contents = ]; span = pass_simple.lua[1:14-1:15]}});
        assign_eq =
        Node.Node {leading_trivia = [];
          trailing_trivia =
          [{ Span.value = (Node.Whitespace " ");
             span = pass_simple.lua[1:17-1:18] }
            ];
          contents = =; span = pass_simple.lua[1:16-1:17]};
        assign_vals =
        (Syntax.SepList1.Mono
           (Syntax.Ref
              Syntax.NDot {
                tbl =
                (Syntax.Ref
                   (Syntax.NVar
                      (Syntax.Var
                         Node.Node {leading_trivia = [];
                           trailing_trivia = []; contents = "test";
                           span = pass_simple.lua[1:18-1:22]})));
                dot =
                Node.Node {leading_trivia = []; trailing_trivia = [];
                  contents = .; span = pass_simple.lua[1:22-1:23]};
                key =
                Node.Node {leading_trivia = [];
                  trailing_trivia =
                  [{ Span.value = (Node.Whitespace " ");
                     span = pass_simple.lua[1:28-1:29] };
                    { Span.value = (Node.LineComment "- @local");
                      span = pass_simple.lua[1:29-1:39] };
                    { Span.value = (Node.Whitespace "\n");
                      span = pass_simple.lua[1:39-2:1] }
                    ];
                  contents = "enter"; span = pass_simple.lua[1:23-1:28]}}))
        })
    ];
  eof =
  Node.Node {leading_trivia = []; trailing_trivia = [];
    contents = end of file; span = pass_simple.lua[2:1-2:1]}
  }